packages/cli/internal/pkg/aws/cdk/progress.go

```diff
@@ -41,19 +50,145 @@ func (p ProgressStream) DisplayProgress(description string) error {
 	return nil
 }
 
-func runProgressBar(ctx context.Context, description string) chan ProgressEvent {
+func ShowExecution(progressEvents []ProgressStream) []Result {
+	var progressResults []*Result
+	var waitGroup sync.WaitGroup
+	waitGroup.Add(len(progressEvents))
+
+	singleStream, progressResults := combineProgressEvents(progressEvents)
+
+	for event := range singleStream {
+		if event.LastOutput != "" {
+			log.Info().Msg(event.LastOutput)
+		}
+	}
+
+	var results []Result
+	for _, progressResult := range progressResults {
+		results = append(results, *progressResult)
+	}
+
+	return results
+}
+
+func updateResultFromStream(stream ProgressStream, progressResult *Result, wait *sync.WaitGroup) {
+	defer wait.Done()
+	var lastEvent ProgressEvent
+
+	for event := range stream {
+		if event.Err != nil {
+			progressResult.Err = event.Err
+		} else {
+			lastEvent = event
+		}
+	}
+
+	progressResult.Outputs = lastEvent.Outputs
+	progressResult.UniqueKey = lastEvent.UniqueKey
+}
+
+func DisplayProgressBar(description string, progressEvents []ProgressStream) []Result {
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	singleStream, progressResults := combineProgressEvents(progressEvents)
+
+	barReceiver := runProgressBar(ctx, description, len(progressEvents))
+	for event := range singleStream {
+		barReceiver <- event
+	}
+
+	var results []Result
+	for _, progressResult := range progressResults {
+		results = append(results, *progressResult)
+	}
+
+	return results
+}
+
+func combineProgressEvents(progressEventChannels []ProgressStream) (ProgressStream, []*Result) {
+	var results []*Result
+	receiver := make(ProgressStream)
+	var waitGroup sync.WaitGroup
+	waitGroup.Add(len(progressEventChannels))
+
+	for _, channel := range progressEventChannels {
+		progressResult := &Result{}
+		results = append(results, progressResult)
+		go sendDataToReceiverAndUpdateResult(channel, progressResult, &waitGroup, receiver)
+	}
+
+	go closeChannelAfterWaitGroup(receiver, &waitGroup)
+
+	return receiver, results
+}
+
+func sendDataToReceiverAndUpdateResult(channel <-chan ProgressEvent, progressResult *Result, waitGroup *sync.WaitGroup, receiver chan ProgressEvent) {
+	defer waitGroup.Done()
+
+	var lastEvent ProgressEvent
+	var stopProcessingEvent = ProgressEvent{
+		CurrentStep: 1,
+		TotalSteps:  1,
+	}
+	for initialChannelOut := range channel {
+		if initialChannelOut.Err != nil {
+			progressResult.Err = initialChannelOut.Err
+
+			stopProcessingEvent.UniqueKey = lastEvent.UniqueKey
+			receiver <- stopProcessingEvent
+
+			progressResult.Outputs = lastEvent.Outputs
+			progressResult.UniqueKey = lastEvent.UniqueKey
+			return
```

Comment:
The thing here is a race condition. We can discuss it offline, but in a nutshell, this last "stopProcessing" event can be missed, because waitGroup is released and program exits BEFORE, it has a chance to process the event.